cut down on number of function calls in the dispatcher by manually inlining functions. fix #43253

bdhirsh opened the pull request:
fixes #43253

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46540 cut down on number of function calls in the dispatcher by manually inlining functions. fix #43253**

Differential Revision: [D24394358](https://our.internmc.facebook.com/intern/diff/D24394358)

bdhirsh committed:
cut down on number of function calls in the dispatcher by manually inlining functions. fix #43253

[ghstack-poisoned]

bdhirsh committed:
Update on "cut down on number of function calls in the dispatcher by manually inlining functions. fix #43253"


Differential Revision: [D24394358](https://our.internmc.facebook.com/intern/diff/D24394358)

[ghstack-poisoned]

ezyang reviewed (approved):
Oh oops, did we fail to actually land this one?

bdhirsh committed:
Update on "cut down on number of function calls in the dispatcher by manually inlining functions. fix #43253"

fixes #43253


Differential Revision: [D24394358](https://our.internmc.facebook.com/intern/diff/D24394358)

[ghstack-poisoned]

bdhirsh commented:
> Oh oops, did we fail to actually land this one?

Yeah :( found it when I was going through my current open PR's

The PR needed to be rebased and I couldn't find the original branch to make the change on- the code changes were small enough that I thought it was less work to just remake the PR. I'll close the old one when the new one lands.
